### PR TITLE
Updated Apple presence

### PR DESCRIPTION
**Pull request by N0chteil**
Proof:
Screenshot_386
Screenshot_387
Screenshot_388
Screenshot_389
Screenshot_390

### CODE REVIEW

**Review by Bas950 — APPROVED** [review]

File: websites/A/Apple/presence.ts [hunk 1]
```diff
@@ -73,6 +74,8 @@ presence.on("UpdateData", async () => {
       largeImageKey: logoArr[setting.logo] || "logo"
     };
 
+    let showElapsed = true;
```
Comment (ririxidev) [inline websites/A/Apple/presence.ts:77]: wow that's the most unecessary thing that I ever seen. Please do it in different way

File: websites/A/Apple/presence.ts [hunk 2]
```diff
@@ -461,9 +464,229 @@ presence.on("UpdateData", async () => {
...
+    } else if(urlpath[1] === 'documentation') {
+      const page = document.querySelector("span.current.item");
+
+      presenceData.details = 'Documentation';
+      presenceData.state = page?.textContent || "Home";
+
+      if (setting.buttons && page) {
+        presenceData.buttons = [
+          {
+            label: `View Page`,
```
Comment (ririxidev) [inline websites/A/Apple/presence.ts:541]:
```suggestion
            label: "View Page",
```

File: websites/A/Apple/presence.ts [hunk 3]
```diff
@@ -461,9 +464,229 @@ presence.on("UpdateData", async () => {
...
+      presenceData.details = document.querySelector("h2.localnav-title>a")?.textContent || "Apple Developer";
+      presenceData.state = document.querySelector("a.localnav-menu-link.link-"+cpage)?.textContent || document.querySelector("span.localnav-menu-link.current")?.textContent || "Other";
+    } else if(urlpath[1] === 'games')
+      presenceData.state = "Games";
+    else if(urlpath[1] === 'forums') {
+      const buttons = {
+        show: false,
+        label: "View Page",
+        url: window.location.href
+      };
```
Comment (ririxidev) [inline websites/A/Apple/presence.ts:571]: I'm literally confused why you have something like `show`. If you don't want to show this, why you just don't delete this const??

File: websites/A/Apple/presence.ts [hunk 4]
```diff
@@ -461,9 +464,229 @@ presence.on("UpdateData", async () => {
...
+        const vid = document.querySelector("video#video"),
+          video_startTime = Date.now(),
+          video_endTime = Math.floor(video_startTime / 1000) - vid.currentTime + vid.duration + 1;
+
+        presenceData.state = document.querySelector("li.supplement.details>h1")?.textContent;
+
+        showElapsed = false;
+
+        if(vid) {
+          presenceData.startTimestamp = video_startTime;
```
Comment (ririxidev) [inline websites/A/Apple/presence.ts:645]:
```suggestion
```

**Review by ririxidev — CHANGES_REQUESTED** [review]

File: websites/A/Apple/presence.ts (hunk 3 above)
Comment (N0chteil) [reply websites/A/Apple/presence.ts:571]: image
So that I don't have this 3 times

Comment (ririxidev) [reply websites/A/Apple/presence.ts:569]: I'm confused af. You only set `show` to false on this const

**Comment by ririxidev** [comment]
also fix deepscan :)

File: websites/A/Apple/presence.ts [hunk 5]
```diff
@@ -461,9 +462,227 @@ presence.on("UpdateData", async () => {
...
+      presenceData.details = document.querySelector("h2.localnav-title>a")?.textContent || "Apple Developer";
+      presenceData.state = document.querySelector("a.localnav-menu-link.link-"+cpage)?.textContent || document.querySelector("span.localnav-menu-link.current")?.textContent || "Other";
+    } else if(urlpath[1] === 'games')
+      presenceData.state = "Games";
+    else if(urlpath[1] === 'forums') {
+      const buttons = {
+        show: false,
+        label: "View Page",
+        url: window.location.href
+      };
```
Comment (Bas950) [inline websites/A/Apple/presence.ts:569]: ???

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/A/Apple/presence.ts (hunk 5 above)
Comment (N0chteil) [reply websites/A/Apple/presence.ts:569]: https://github.com/PreMiD/Presences/pull/3742#discussion_r633587692

Comment (ririxidev) [reply websites/A/Apple/presence.ts:569]: and check what i wrote about it. It's unnecessary

**Review by Bas950 — APPROVED** [review]

**Pull request merged**